parlai/core/teachers.py

```diff
@@ -589,55 +589,58 @@ def _read_episode(self, data_loader):
                 episode = []
                 last_cands = None
 
-            # intern all strings so we don't store them more than once
-            new_entry = []
-            if len(entry) > 0:
-                # process text if available
-                if entry[0] is not None:
-                    new_entry.append(sys.intern(entry[0]))
-                else:
-                    new_entry.append(None)
-                if len(entry) > 1:
-                    # process labels if available
-                    if entry[1] is None:
-                        new_entry.append(None)
-                    elif hasattr(entry[1], '__iter__') and type(entry[1]) is not str:
-                        # make sure iterable over labels, not single string
-                        new_entry.append(tuple(sys.intern(e) for e in entry[1]))
+            if isinstance(entry, dict):
+                episode.append(entry)
+            else:
+                # intern all strings so we don't store them more than once
```

Comment:
I didn't realize there was a call to sys.intern here -- I actually wasn't familiar with that call. I wonder cheap that one time cost is and how much it gains us later on.